src/test/coins_tests.cpp

```diff
@@ -916,6 +916,135 @@ BOOST_AUTO_TEST_CASE(coins_coinbase_spends)
+
+            // 1/10 times create a coinbase
+            if (insecure_rand() % 10 == 0 || coinbaseids.size() < 10) {
+                // 1/100 times create a duplicate coinbase
+                if (insecure_rand() % 10 == 0 && coinbaseids.size()) {
+                    std::map<uint256, CAmount>::iterator coinbaseIt = coinbaseids.lower_bound(GetRandHash());
+                    if (coinbaseIt == coinbaseids.end()) {
+                        coinbaseIt = coinbaseids.begin();
+                    }
+                    //Use same random value to have same hash and be a true duplicate
+                    tx.vout[0].nValue = coinbaseIt->second;
+                    assert(tx.GetHash() == coinbaseIt->first);
+                    duplicateids.insert(coinbaseIt->first);
+                }
+                else {
+                    coinbaseids[tx.GetHash()] = tx.vout[0].nValue;
+                }
+                assert(CTransaction(tx).IsCoinBase());
+            }
+            // 9/10 times create a regular tx
+            else {
+                uint256 prevouthash;
+                // equally likely to spend coinbase or non coinbase
+                std::set<uint256>::iterator txIt = alltxids.lower_bound(GetRandHash());
+                if (txIt == alltxids.end()) {
+                    txIt = alltxids.begin();
+                }
+                prevouthash = *txIt;
+
+                // Construct the tx to spend the coins of prevouthash
+                tx.vin[0].prevout.hash = prevouthash;
+                tx.vin[0].prevout.n = 0;
+
+                // Update the expected result of prevouthash to know these coins are spent
+                CCoins& oldcoins = result[prevouthash];
+                oldcoins.Clear();
+
+                // It is of particular importance here that once we spend a coinbase tx hash
+                // it is no longer available to be duplicated (or spent again)
+                // BIP 34 in conjunction with enforcing BIP 30 (at least until BIP 34 was active)
+                // results in the fact that no coinbases were duplicated after they were already spent
+                alltxids.erase(prevouthash);
+                coinbaseids.erase(prevouthash);
+
+                // The test is designed to ensure spending a duplicate coinbase will work properly
+                // if that ever happens and not resurrect the previously overwritten coinbase
+                if (duplicateids.count(prevouthash))
+                    spent_a_duplicate_coinbase = true;
+
+                assert(!CTransaction(tx).IsCoinBase());
+            }
+            // Track this tx to possibly spend later
+            alltxids.insert(tx.GetHash());
+
+            // Update the expected result to know about the new output coins
+            CCoins &coins = result[tx.GetHash()];
+            coins.FromTx(tx, height);
+
+            CValidationState dummy;
+            UpdateCoins(tx, *(stack.back()), height);
```

Comment:
Note: In zcash,```CValidationState dummy;``` is not used. In bitcoin, the call is ```UpdateCoins(tx, dummy, *(stack.back()), height);``` I presume this is being left to minimize future merge conflicts.

Reply:
It's an artifact of the PR this change is from being merged upstream before bitcoin/bitcoin#7976, which removes the argument, and which we have already merged. They in fact removed `dummy` in that PR; I will do so in this PR.